hyperapp.d.ts

```diff
@@ -74,9 +83,7 @@ export type ActionsType<State, Actions> = {
  * @returns A VNode tree.
  * @memberOf [App]
  */
-export interface View<State, Actions> {
-  (state: State, actions: Actions): VNode<object>
-}
+type View<State, Actions> = (state: State, actions: Actions) => VNode<object>
```

Comment:
I find `type` much better as it's a function not an object. No need to export but I should have done for consistency :(

Reply:
Don't do this. It disallows declaration merging of interfaces/types. 

You can do this (below) with interfaces, but not with types because the names need to be unique. Don't use _types_ for typings that are exposed to the world. Generally, always use interfaces, they're a bit more flexible, unless you want it to be unique and 'protected' in some way

```typescript
interface Box {
  size: number;
}
interface Box {
  height: number;
  width: number;
}
```

Reply:
Interesting. That's new to me. I read the typescript manual to say that interfaces are for objects not functions (they have separate sections) . Given this, using an interface to declare a function signature seems plain weird. It's not a interface that contains a function - it IS a function. Or am I missing somehting ?

https://www.typescriptlang.org/docs/handbook/interfaces.html
https://www.typescriptlang.org/docs/handbook/functions.html

To my mind functions do things to the values that Interfaces define. These are not the same interfaces as C++ or C#. But Interfaces can have function members of course. But that's not the case with View - it's a function nothing more.

Your example is clearly an object not a function

Reply:
It's still limiting the end user to modify and extend the typing by using the concept of declaration merging. Generally, when typing a library, always use interfaces, since they're more flexible for the end user.

> using an interface to declare a function signature **seems plain weird**

I wonder what the creators of Typescript would feel about this. There's clearly a reason to be able to define functions as interfaces. To you it might feel *plain weird*, because you don't fully understand the reasoning behind it.

Reply:
> To you it might feel plain weird, because you don't fully understand the reasoning behind it.

That's often the case, for sure. However I can't find that reasoning in the typescript docs or ever seen other examples.

Reply:
I agree that TypeScript recommends using interface, but I still think that type should be used in this case. Because hyperapp's view is expected to behave as a simple function.

The reason interface is preferable to type is that users can add properties and methods by using it. But is hyperapp's view like something the user wants to add properties and methods? I do not think so.

Reply:
I'm obviously going to have to get to understand this better. to my mind the previous declaration with interface was a object with a function member. I'm obviously got that wrong. But agree in this case would never catenate extra items - it's a stand alone function not an interface

Reply:
Now I understand the different between a function interface and an object with a function member! It's a pure declaration without a name!

However, while it is not going to get extended,  there is one advantage of interface over type alias - the "View" name will appear in error messages not just the type details.